packages/govern-console/src/containers/DAOSettings/DAOSettings.tsx

```diff
@@ -244,231 +233,230 @@ const DaoSettings: React.FC<DaoSettingFormProps> = () => {
-                    onChange={onChange}
-                    value={value.toString()}
-                    wide
-                    placeholder={'10.0'}
-                    status={!!error ? 'error' : 'normal'}
-                    error={error ? error.message : null}
-                  />
-                )}
-              />
-              <Controller
-                name="daoConfig.challengeDeposit.token"
-                control={control}
-                defaultValue=""
-                rules={{
-                  required: 'This is required.',
-                  validate: async (value) => {
-                    const v = await validateToken(value, provider);
-                    if (v !== true) {
-                      return v;
-                    }
+                  let { decimals } = await getTokenInfo(value, provider);
+                  decimals = decimals || 0;
+
+                  setScheduleDecimals(decimals);
+                  await trigger('daoConfig.scheduleDeposit.amount');
+                },
+              }}
+              render={({ field: { onChange, value }, fieldState: { error } }) => (
+                <TextInput
+                  title="Schedule execution collateral"
+                  onChange={onChange}
+                  value={value}
+                  wide
+                  placeholder={'0x'}
+                  status={!!error ? 'error' : 'normal'}
+                  error={error ? error.message : null}
+                />
+              )}
+            />
+            <Controller
+              name="daoConfig.scheduleDeposit.amount"
+              control={control}
+              defaultValue={''}
+              rules={{
+                required: 'This is required.',
+                validate: (value) => validateAmountForDecimals(value, scheduleDecimals),
+              }}
+              render={({ field: { onChange, value }, fieldState: { error } }) => (
+                <TextInput
+                  title={'Token amount'}
+                  type="number"
+                  onChange={onChange}
+                  value={value.toString()}
+                  wide
+                  placeholder={'10.0'}
+                  status={!!error ? 'error' : 'normal'}
+                  error={error ? error.message : null}
+                />
+              )}
+            />
+            <Controller
+              name="daoConfig.challengeDeposit.token"
+              control={control}
+              defaultValue=""
+              rules={{
+                required: 'This is required.',
+                validate: async (value) => {
+                  const v = await validateToken(value, provider);
+                  if (v !== true) {
+                    return v;
+                  }
 
-                    let { decimals } = await getTokenInfo(value, provider);
-                    decimals = decimals || 0;
+                  let { decimals } = await getTokenInfo(value, provider);
```

Comment:
hmmm... we need to clean up some code here as there is local function `getTokenInfo()` and `getToken()` from the govern library. I used the one from govern...

Reply:
I agree